### PR TITLE
Store: Add support for universal analytics for stores

### LINKED ISSUE #17105: Store: Google Analytics for Stores on WordPress.com (Dev)
Updated: Add new settings to our Google Analytics offering to support:

* IP address anonymization on/off

And for sites with WooCommerce active:

* eCommerce tracking on/off
* Add to cart event tracking on/off
* Enhanced eCommerce tracking on/off
* Remove from cart event tracking on/off
* Product impression event tracking on/off
* Product click event tracking on/off
* Product detail view tracking on/off
* Checkout process initiated event tracking on/off

Note: We already have a means of setting the Google Tracking ID (and that seems to be the only setting) in Calypso at https://wordpress.com/settings/traffic/{domain} under Google Analytics

Design issue: #17060 

cc @beaulebens @kellychoffman

### PR DESCRIPTION
**Pull request by allendav**
Closes #17105

To test:
* First, comment out these lines to avoid a bazillion annoying warnings due to another component in the settings > traffic page

https://github.com/Automattic/wp-calypso/blob/b855b7a8a7e9f6a24e4fc87e8fe251cfd10603fd/client/boot/app.js#L5-L7

https://github.com/Automattic/wp-calypso/issues/18592#issuecomment-346142563

* First, let's make sure you have a fresh clean Jetpack installed and connected
* Deactivate and delete the Jetpack plugin if you are running it
* Install the Jetpack plugin from the .org repository and activate and connect it
* Install the Jetpack beta plugin from https://jetpack.com/download-jetpack-beta/ and activate it
* Note: If you have WooCommerce Services 1.9.0 installed, you will get a PHP Notice when you deactivate Jetpack: Use of undefined constant JETPACK__VERSION. This can be ignored - it will be fixed in the next version of WooCommerce services - https://github.com/Automattic/woocommerce-services/issues/1249
* Go to /wp-admin/admin.php?page=jetpack-beta and activate the RC `5.6-beta2` release
* Note the Jetpack version you get on wp-admin/plugins.php and make sure it is 5.6 or higher
* Install and activate the Jetpack Google Analytics Helper plugin https://github.com/automattic/jetpack-google-analytics-helper
* Install this branch and navigate to http://calypso.localhost:3000/settings/traffic/{DOMAIN}
* Enter a valid Google Analytics Tracking ID and hit Save Settings (leave all the other settings off for now)
* Verify you are told settings saved successfully
* Open https://{DOMAIN}/wp-admin/tools.php?page=jpgah
* Make sure the settings there match what you see at http://calypso.localhost:3000/settings/traffic/{DOMAIN}
* Open https://{DOMAIN}
* Make sure if you view the page source, you see your tracking ID there
* Modify a few other settings on http://calypso.localhost:3000/settings/traffic/{DOMAIN} and save them
* Refresh https://{DOMAIN}/wp-admin/tools.php?page=jpgah
* Make sure the settings there match what you see at http://calypso.localhost:3000/settings/traffic/{DOMAIN}
* Bonus: Use the testing instructions on https://github.com/Automattic/jetpack/pull/8218 as a guide to ensure that your Google Analytics dashboard gets the expected data.

Wut It Looks Like:

screen shot 2017-11-21 at 3 51 37 pm

screen shot 2017-11-21 at 3 52 08 pm

### CODE REVIEW

File: client/my-sites/site-settings/form-analytics-stores.jsx [hunk 1]
```diff
@@ -53,13 +55,57 @@ class FormAnalyticsStores extends Component {
...
 	render = () => {
 		const { disabled, translate } = this.props;
 
+		// TODO config flag it
+
 		return (
 			<div>
 				<FormLegend>{ translate( 'Basic store analytics' ) }</FormLegend>
 				{ this.renderBasicSettings( disabled ) }
+				<FormLegend>{ translate( 'Advanced store analytics' ) }</FormLegend>
```
Comment (a8ci18n) [inline client/my-sites/site-settings/form-analytics-stores.jsx]: Hi! I've found a possible matching string that has already been translated **16** times:
`translate( 'Basic store analytics' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/site-settings/form-analytics-stores.jsx [hunk 2]
```diff
@@ -53,13 +70,53 @@ class FormAnalyticsStores extends Component {
...
+
 	render = () => {
 		const { disabled, translate } = this.props;
+		const showEnhanced = config.isEnabled( 'jetpack/google-analytics-for-stores-enhanced' );
 
 		return (
 			<div>
 				<FormLegend>{ translate( 'Basic store analytics' ) }</FormLegend>
 				{ this.renderBasicSettings( disabled ) }
+				{ showEnhanced && <FormLegend>{ translate( 'Enhanced store analytics' ) }</FormLegend> }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/form-analytics-stores.jsx]: Hi! I've found a possible matching string that has already been translated **16** times:
`translate( 'Basic store analytics' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/site-settings/form-analytics-stores.jsx [hunk 3]
```diff
@@ -53,13 +69,52 @@ class FormAnalyticsStores extends Component {
...
 	};
 
+	renderEnhancedSettings = disableAll => {
+		const { translate } = this.props;
+		const settings = [
+			{
+				key: 'enh_ec_tracking',
+				label: translate( 'Enable Enhanced eCommerce' ),
+				explanation: translate(
+					'Before enabling, turn on enhanced eCommerce in your Google Analytics dashboard.'
```
Comment (justinshreve) [inline client/my-sites/site-settings/form-analytics-stores.jsx]: Can we link part of this (or something here) to https://support.google.com/analytics/answer/6032539?

**Comment by kellychoffman** [comment]
Lookin' good. 

> The code is looking good to me. I think we should include a link somewhere about how to turn on enhanced eCommerce though.

Agree. Can we add the links that were in the design? 
ga-in-jetpack-settings

Anonymize toggle link: https://support.google.com/analytics/answer/2763052

Advanced ecommerce link: https://support.google.com/analytics/answer/6032539

**Comment by kellychoffman** [comment]
One more nitpick: The toggles themselves look to be a few pixels too low. If they were moved up a bit, they'd vertically align with the labels.

**Comment by allendav** [comment]
Test instructions updated now that Jetpack 5.6 alpha includes the updated module.

File: client/my-sites/site-settings/form-analytics-stores.jsx (hunk 3 above)
Comment (allendav) [reply client/my-sites/site-settings/form-analytics-stores.jsx]: Fixed in https://github.com/Automattic/wp-calypso/pull/19888/commits/0701418362cd26b4dba6cf4c9a7edad07d387b95

**Comment by allendav** [comment]
@kellychoffman wrote:

> Can we add the links that were in the design?

> The toggles themselves look to be a few pixels too low. If they were moved up a bit, they'd vertically align with the labels.

Both fixed in https://github.com/Automattic/wp-calypso/pull/19888/commits/0701418362cd26b4dba6cf4c9a7edad07d387b95

**Comment by allendav** [comment]
Looks like this now

updated-analytics-settings

**Comment by allendav** [comment]
@justinshreve wrote:

> We also need to add a feature flag to desktop now.

Fixed in https://github.com/Automattic/wp-calypso/pull/19888/commits/4c0598fdb5574a2301062e68df2825312663af4b - added the goodies for the previous work too

**Comment by allendav** [comment]
@oskosk @dereksmart @justing - I just realized that `5.6-alpha` is not the most appropriate version check here - since that version appeared on jetpack master 20 days ago.

This PR currently uses that version check to decide whether or not to attempt to use version 1.4 of the settings endpoint for the site - and that was just merged to jetpack only a few days ago.

So...

If I merge this as is, then sites running older alphas will not work correctly with the Calypso settings pages.

So...

I think I should NOT commit this PR with that version check, but wait for the next "tag" of Jetpack, then update this PR with that tag and then we're good.

**Pull request merged**